### ISSUE TITLE
How to custom System.Serializable ?

### ISSUE THREAD

**Issue by DavidBabel** [Kinwailo/Stackable-Decorator#10]
I got this code in my project
```csharp
  [System.Serializable]
  public class Size {
    public int width;
    public int height;
    // ...
```
And it display this way :
![image](https://user-images.githubusercontent.com/5599375/43601455-7f691008-968d-11e8-9e73-7cb4996ab80a.png)

Can i use in any way your decorators to display them nicely ? For example, without the Expandable part would be a start.

**Comment by ShivanshuKantPrasad** [comment]
I Created this attribute `[NoFoldout]` which display this way:
![capture](https://user-images.githubusercontent.com/17960975/43672484-58371c24-97cc-11e8-93b9-979d2036d476.PNG)

If this is what you want here is my code (it's a bit hacky but it works)
`
using UnityEditor;
using UnityEngine;
using StackableDecorator;
using System.Linq;

public class NoFoldout : StackableFieldAttribute {

    public override void OnGUI(Rect position, SerializedProperty property, GUIContent label, bool includeChildren) {
		if (!includeChildren || !HasVisibleChildFields(property))
                DefaultPropertyField(position, property, label);
            else
            {
                var iconSize = EditorGUIUtility.GetIconSize();
                bool enabled = GUI.enabled;
                int indentLevel = EditorGUI.indentLevel;

                var propertyHeights = m_PropertyHeights.Get(property.propertyPath);
                int index = 0;

				EditorGUI.PrefixLabel(position, label);

                var prop = property.Copy();
                var depth = prop.depth;
                EditorGUI.indentLevel = prop.depth + (indentLevel - depth);
                position.height = EditorGUI.GetPropertyHeight(property.propertyType, label);
                bool enter = HasVisibleChildFields(prop);
                if (!enter) return;

                position.y += position.height + 2f;
                while (prop.NextVisible(enter) && prop.depth > depth)
                {
                    enter = false;
                    prop.GetFieldInfos(s_FieldInfos);
                    if (s_FieldInfos[s_FieldInfos.Count - 1].GetCustomAttributes(typeof(InGroupAttribute), false).Any())
                        continue;

                    EditorGUI.indentLevel = prop.depth + (indentLevel - depth);
                    if (propertyHeights.Any())
                        position.height = propertyHeights[index];
                    else
                        position.height = EditorGUI.GetPropertyHeight(prop, null, false);
                    index++;

                    EditorGUI.BeginChangeCheck();
                    enter = EditorGUI.PropertyField(position, prop, null, false) && HasVisibleChildFields(prop);
                    if (EditorGUI.EndChangeCheck()) break;
                    position.y += position.height + 2f;
                }
                GUI.enabled = enabled;
                EditorGUIUtility.SetIconSize(iconSize);
                EditorGUI.indentLevel = indentLevel;
            }
	}
}
`

**Comment by DavidBabel** [comment]
Seems great, but maybe you adapt a part of your library too ?

Because i get errors like "** is inaccessible due to its protection level" fo both **m_PropertyHeights** and **s_FieldInfos**. Maybe i forgot something ?

![image](https://user-images.githubusercontent.com/5599375/43674621-581d08a6-97d7-11e8-8804-503737959874.png)

**Comment by ShivanshuKantPrasad** [comment]
Yes I changed those items in the base class from private to protected. Sorry, that's why i said it's hacky.
